flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/translators/TwoInputTransformationTranslator.java

```diff
@@ -46,7 +47,11 @@
                 transformation.getStateKeySelector1() != null
                         && transformation.getStateKeySelector2() != null;
         if (isKeyed) {
```

Comment:
I think we should change it so that we apply sorting if any of the inputs is keyed.

Reply:
I think we have to yes. I shied away from it here because I would have to add a test that uses the low-level API because right now it's not possible for users to create such transforms.

Could you maybe whip together a test based on the existing test code you already had? We add it here and I change it to correctly forward the requirements?

Reply:
sure, I will prepare one

Reply:
I was thinking of tests like those: https://github.com/dawidwys/flink/blob/mix-keyed-2/flink-tests/src/test/java/org/apache/flink/api/datastream/DataStreamBatchExecutionITCase.java

Reply:
Sorry, I also got started on it and should have notified you. 🙈